pkg/dashboard/dashboard.go

```diff
@@ -52,6 +66,47 @@ func Dashboard(opts Options) http.Handler {
 			return
 		}
 
+		if costPerCPU != "" && costPerGB != "" {
+			costPerCPUFloat, _ := strconv.ParseFloat(costPerCPU, 64)
+			costPerGBFloat, _ := strconv.ParseFloat(costPerGB, 64)
+
+			for _, n := range vpaData.Namespaces {
+				for _, w := range n.Workloads {
+					for k, c := range w.Containers {
+						cpuCost := costPerCPUFloat * (c.Requests.Cpu().AsApproximateFloat64() + c.Limits.Cpu().AsApproximateFloat64()) / 2
+						memCost := costPerGBFloat * (ConvertToGB(c.Requests.Memory().Value()) + ConvertToGB(c.Limits.Memory().Value())) / 2
+						topNumber := toFixed(cpuCost+memCost, 4)
+
+						cpuCostRecommended1 := costPerCPUFloat * c.Target.Cpu().AsApproximateFloat64()
+						memCosttRecommended1 := costPerGBFloat * ConvertToGB(c.Target.Memory().Value())
+
+						cpuCostRecommended2 := costPerCPUFloat * (c.LowerBound.Cpu().AsApproximateFloat64() + c.UpperBound.Cpu().AsApproximateFloat64()) / 2
+						memCosttRecommended2 := costPerGBFloat * (ConvertToGB(c.LowerBound.Memory().Value()) + ConvertToGB(c.UpperBound.Memory().Value())) / 2
+
+						lowerNumber1 := toFixed(topNumber-(cpuCostRecommended1+memCosttRecommended1), 4)
+						lowerNumber2 := toFixed(topNumber-(cpuCostRecommended2+memCosttRecommended2), 4)
+
+						if lowerNumber1 < 0 {
+							c.LowerNumberInt1 = -1
+						} else if lowerNumber1 > 0 {
+							c.LowerNumberInt1 = 1
+						}
+						if lowerNumber2 < 0 {
+							c.LowerNumberInt2 = -1
+						} else if lowerNumber2 > 0 {
+							c.LowerNumberInt2 = 1
+						}
+
+						c.TopNumber = topNumber
+						c.LowerNumber1 = math.Abs(lowerNumber1)
+						c.LowerNumber2 = math.Abs(lowerNumber2)
+
+						w.Containers[k] = c
+					}
+				}
+			}
+		}
```

Comment:
It would be really nice if this was split into multiple functions in order to reduce the overall complexity of this section. As it is now, it's very confusing, and I am struggling to understand what it's supposed to be doing. Additionally, if we split them out, then we can test these functions, which will add even more clarity.

Reply:
I updated the code. Thank you!